v0/destinations/fbpixel/transform.js

```diff
@@ -0,0 +1,431 @@
+  return Number((revenue || 0).toFixed(2));
+}
+
+function getContentType(message, defaultValue, categoryToContent) {
+  const { options } = message;
+  if (options && options.contentType) {
+    return [options.contentType];
+  }
+
+  let { category } = message.properties;
+  if (!category) {
+    const { products } = message.properties;
+    if (products && products.length) {
+      category = products[0].category;
+    }
+  }
+  if (category) {
+    if (categoryToContent === undefined) {
+      categoryToContent = [];
+    }
+    const mapped = categoryToContent;
+    const mappedTo = mapped.reduce((filtered, map) => {
+      if (map.from === category) {
+        filtered = map.to;
+      }
+      return filtered;
+    }, "");
+    if (mappedTo.length) {
+      return mappedTo;
+    }
+  }
+  return defaultValue;
+}
+
+function handleOrder(message, categoryToContent) {
+  const { products } = message.properties;
+  const value = formatRevenue(message.properties.revenue);
+  const contentType = getContentType(message, "product", categoryToContent);
+  const contentIds = [];
+  const contents = [];
+  const { category } = message.properties;
+
+  for (let i = 0; i < products.length; i += 1) {
+    const pId = products[i].product_id || products[i].sku || products[i].id;
+    contentIds.push(pId);
+    const content = {
+      id: pId,
+      quantity: products[i].quantity,
+      item_price: products[i].price
+    };
+    contents.push(content);
+  }
+  return {
+    content_category: category,
+    content_ids: contentIds,
+    content_type: contentType,
+    currency: message.properties.currency,
+    value,
+    contents,
+    num_items: contentIds.length
+  };
+}
+function handleProductListViewed(message, categoryToContent) {
+  let contentType;
+  const contentIds = [];
+  const contents = [];
+  const { products } = message.properties;
+  if (Array.isArray(products)) {
+    products.forEach(function(product) {
+      const productId = product.product_id;
+      if (productId) {
+        contentIds.push(productId);
+        contents.push({
+          id: productId,
+          quantity: message.properties.quantity
+        });
+      }
+    });
+  }
+
+  if (contentIds.length) {
+    contentType = "product";
+  } else {
+    contentIds.push(message.properties.category || "");
+    contents.push({
+      id: message.properties.category || "",
+      quantity: 1
+    });
+    contentType = "product_group";
+  }
+  return {
+    content_ids: contentIds,
+    content_type: getContentType(message, contentType, categoryToContent),
+    contents
+  };
+}
+
+function handleProduct(message, categoryToContent, valueFieldIdentifier) {
+  const useValue = valueFieldIdentifier === "properties.value";
+  const contentIds = [
+    message.properties.product_id ||
+      message.properties.id ||
+      message.properties.sku ||
+      ""
+  ];
+  const contentType = getContentType(message, "product", categoryToContent);
+  const contentName =
```

Comment:
let's create a EcoomMappings.js (like GenericMappings.json) and put these mappings there and utility methods like getProductId, getProductCategory, getProductPrice(value/price/fieldName), getProductQuantity etc.
It has two advantages IMO:
1. we can use get library, that handles empty properties and return default value 
2. we can use this as our common ecomm spec